contribs/src/main/java/com/netflix/conductor/contribs/queue/kafka/KafkaObservableQueue.java

```diff
@@ -0,0 +1,230 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.conductor.contribs.queue.kafka;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.KafkaException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.netflix.conductor.core.config.Configuration;
+import com.netflix.conductor.core.events.queue.Message;
+import com.netflix.conductor.core.events.queue.ObservableQueue;
+import com.netflix.conductor.core.execution.ApplicationException;
+
+import rx.Observable;
+import rx.Observable.OnSubscribe;
+
+/**
+ * Reads the properties with prefix 'kafka.producer.', 'kafka.consumer.'
+ * and 'kafka.' from the provided configuration. Initializes a producer 
+ * and consumer based on the given value. Queue name is driven from the 
+ * workflow. 
+ * @author preeth
+ *
+ */
+public class KafkaObservableQueue implements ObservableQueue {
+
+  private static final Logger logger = LoggerFactory.getLogger(KafkaObservableQueue.class);
+
+  private static final String QUEUE_TYPE = "kafka";
+
+  private static final String KAFKA_PRODUCER_PREFIX = "kafka.producer.";
+
+  private static final String KAFKA_CONSUMER_PREFIX = "kafka.consumer.";
+
+  private static final String KAFKA_PREFIX = "kafka.";
+
+  private String queueName;
+
+  private int pollTimeInMS;
+
+  private int longPollTimeout;
+
+  private int pollCount;
+
+  private KafkaProducer<String, String> producer;
+
+  private KafkaConsumer<String, String> consumer;
+
+  @Inject
+  public KafkaObservableQueue(String queueName, Configuration config) {
+
+    this.queueName = queueName;
+    this.pollTimeInMS = config.getIntProperty("workflow.dyno.queues.pollingInterval", 100);
+    this.pollCount = config.getIntProperty("workflow.dyno.queues.pollCount", 10);
+    this.longPollTimeout = config.getIntProperty("workflow.dyno.queues.longPollTimeout", 1000);
+    // Init Kafka producer and consumer properties
+    Properties producerProps = new Properties();
+    Properties consumerProps = new Properties();
+    consumerProps.put("group.id", queueName + "_group");
+    int randomId = new Random().nextInt(100);
+    consumerProps.put("client.id", queueName + "_consumer_"+randomId);
```

Comment:
Use config.getServerId() instead of randomId.

Reply:
done.